File: 2025/day_6/day6b.py
# https://adventofcode.com/2025/day/6
import math

# load input file
use_example = False
with open("input6.txt" if not use_example else "example6.txt", "r") as file:
	lines = file.read().splitlines()

ops = [(i,ch) for i,ch in enumerate(lines.pop()) if ch in "+*"]
lengths = [b[0] - a[0] - 1 for a, b in zip(ops, ops[1:])]
lengths.append(len(lines[0]) - ops[-1][0])

summed = 0
for (idx, op), d in zip(ops, lengths):
	vals = [
		int("".join(line[idx+i] for line in lines))
		for i in range(d)
	]
	summed += sum(vals) if op == "+" else math.prod(vals)

print(summed)
# 11950004808442

File: 2025/day_6/day6b.js
// https://adventofcode.com/2025/day/6
const fs = require('fs');

const use_example = false;
const input = fs.readFileSync(`./${use_example ? "example" : "input"}6.txt`, 'utf8');

const lines = input.split("\n").filter(l => l != "");
let ops = lines.pop();

let offset = 0, columns = [];
while (offset < ops.length) {
	const op = ops.charAt(offset);
	const index = ops.slice(offset + 1).search(/\+|\*/);
	const len = index === -1 ? ops.length - offset : index;

	const vals = [];
	for (let i = 0; i < len; i++) {
		vals.push(lines.reduce((p, c) => { return p + c[offset + i]; }, ""));
	}

	columns.push({ op, vals: vals.map(v => parseInt(v)) });
	offset += len + 1;
}

const results = columns.map(col => {
	const op = col.op;
	return col.vals.reduce((p, c) => { return op == "+" ? p + c : p * c; }, op == "+" ? 0 : 1);
});

const sum = results.reduce((p, c) => { return p + c; }, 0);
console.log(sum);
//11950004808442